ls1intum/Artemis
# Feature/global/cacheable images

### Checklist

### Motivation and Context



We often reload images from the server, even though they never change.
If we cache them, this would reduce the server load.


### Description


We use `ngx-cacheable` for caching the images, `blob-utils` for serializing/deserializing blobs from/to LocalStorage.
The cache will be emptied when the LocalStorage is cleared and on login/logout.

~The size of the LocalStorage is [limited to 5-10MB](https://flaviocopes.com/web-storage-api/#storage-size-limits). When we calculate with an image size of 30kb-50kb, the storage would be full with 100 images in the worst case.
**I limited the cache to 30, because we still have to calculate with larger images and other data in our LocalStorage.**
An alternative would be to store the images in the SessionStorage, but this would mean the caching is lost after closing the browser (not just after closing the tab however!).~

=> :information_source: We decided to use the LocalStorage for the small course icons, and for all other images the SessionStorage.

### Steps for Testing


1. Log in to ArTEMiS
2. Go to course overview for a course that has a course icon
3. Open the course => check in the network tab that the image was loaded
4. Go back and then open the course again => check that the image was retrieved from cache

### Screenshots
